x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -390,6 +414,38 @@ public Object value(Aggregation agg, Map<String, String> fieldTypeMap, String lo
         }
     }
 
+    static class GeoLineAggExtractor implements AggValueExtractor {
+
+        @Override
+        public Object value(Aggregation aggregation, Map<String, String> fieldTypeMap, String lookupFieldPrefix) {
+            assert aggregation instanceof InternalAggregation
+                && ((InternalAggregation) aggregation).getWriteableName()
+                    .equals(TransformAggregations.AggregationType.GEO_LINE.getName()) : "Unexpected type ["
+                        + aggregation.getClass().getName()
+                        + "] for aggregation ["
+                        + aggregation.getName()
+                        + "]";
+            try (XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()) {
+                xContentBuilder.startObject();
+                ((InternalAggregation) aggregation).doXContentBody(xContentBuilder, ToXContent.EMPTY_PARAMS);
+                xContentBuilder.endObject();
+                Map<String, Object> aggMap = XContentHelper.convertToMap(
+                    BytesReference.bytes(xContentBuilder),
+                    false,
+                    XContentType.JSON
+                ).v2();
+                // If `geometry` doesn't exist, then it should just be null
+                return aggMap.get("geometry");
```

Comment:
This is **ROUGH** I would much rather have the `InternalGeoLine` agg in xpack core. That way we can cast the object and get stuff directly.